edumap/app/models/lesson.rb

```diff
@@ -35,7 +35,15 @@ class Lesson < ActiveRecord::Base
   scope :with_created_at_gte, -> (ref_date) { where('created_at >= ?', ref_date) }
 
   scope :with_standard, -> (standards) { with_association(:standards, standards) }
-  scope :with_grade, -> (levels) { with_association(:levels, levels) }
+  scope :with_grade, -> (level) do
+    grade_at_most(level).grade_at_least(level)
```

Comment:
@hwayne: Agreed. Refactored this.

We don't need to group by lesson ID because the main query has no joins.

The way I originally suggested doing this won't work, because ActiveRecord applies all the conditions to the entire join. This is the only way I can think of to do it without using raw SQL, since ActiveRecord doesn't support table aliases as far as I could find out.